config/routes.rb

```diff
@@ -7,7 +7,7 @@
   get '/auth/google_oauth2/callback', to: 'sessions#create'
 
   get '/dashboard', to: "users#show"
-  resources :gardens, only: [:new, :create, :show], shallow: true do
+  resources :gardens, only: [:new, :create, :show, :edit, :update], shallow: true do
```

Comment:
I think it might be time to consider doing the except rather than only for gardens